packages/frontend/src/providers/ExplorerProvider.tsx

```diff
@@ -1105,14 +1106,28 @@ export const ExplorerProvider: FC<{
         });
     }, [mutateAsync, state.shouldFetchResults]);
 
-    const clear = useCallback(async () => {
+    const clearExplore = useCallback(async () => {
         dispatch({
             type: ActionType.RESET,
             payload: defaultState,
         });
         resetQueryResults();
     }, [resetQueryResults]);
 
+    const clearQuery = useCallback(async () => {
+        dispatch({
+            type: ActionType.RESET,
+            payload: {
+                ...defaultState,
+                unsavedChartVersion: {
+                    ...defaultState.unsavedChartVersion,
+                    tableName: unsavedChartVersion.tableName,
+                },
+            },
+        });
+        resetQueryResults();
+    }, [resetQueryResults, unsavedChartVersion.tableName]);
```

Comment:
Hi @ZeRego, thanks for the code review!

You suggested renaming original `clear` because its use case was different, but looking at its usage I thought that `clear` is a good name 🤔 

I didn't find a more suitable name 😅, so in the end I renamed the original one to `clearExplore` and create the suggested one as `clearQuery`. Let me know what you think about!